src/Acuminator/Acuminator.Analyzers/StaticAnalysis/DacKeyFieldDeclaration/KeyFieldDeclarationFix.cs

```diff
@@ -0,0 +1,134 @@
+
+		public override  Task RegisterCodeFixesAsync(CodeFixContext context)
+		{
+			return Task.Run(() =>
+			{
+				var diagnostic = context.Diagnostics.FirstOrDefault(d => d.Id == Descriptors.PX1055_DacKeyFieldBound.Id);
+
+				context.CancellationToken.ThrowIfCancellationRequested();
+
+				if (diagnostic == null)
+					return;
+
+				string codeActionIdentityKeyName = nameof(Resources.PX1055Fix1).GetLocalized().ToString();
+				CodeAction codeActionIdentityKey = CodeAction.Create(codeActionIdentityKeyName,
+																	cToken => RemoveKeysFromFieldsAsync(context.Document,
+																										context,
+																										cToken, 
+																										diagnostic, 
+																										editIdentityAttribute:false),
+																	equivalenceKey: codeActionIdentityKeyName);
+
+				string codeActionBoundKeysName = nameof(Resources.PX1055Fix2).GetLocalized().ToString();
+				CodeAction codeActionBoundKeys = CodeAction.Create(	codeActionBoundKeysName,
+																	cToken => RemoveKeysFromFieldsAsync(context.Document, 
+																										context,
+																										cToken, 
+																										diagnostic, 
+																										editIdentityAttribute: true),
+																	equivalenceKey: codeActionBoundKeysName);
+
+				string codeActionRemoveIdentityColumnName = nameof(Resources.PX1055Fix3).GetLocalized().ToString();
+				CodeAction codeActionRemoveIdentityColumn = CodeAction.Create(codeActionRemoveIdentityColumnName,
+																				cToken => RemoveKeysFromFieldsAsync(context.Document, 
+																													context, 
+																													cToken, 
+																													diagnostic,
+																													removeIdentityAttribute: true),
+																				equivalenceKey: codeActionRemoveIdentityColumnName);
+
+				context.RegisterCodeFix(codeActionIdentityKey, context.Diagnostics);
+				context.RegisterCodeFix(codeActionBoundKeys, context.Diagnostics);
+				context.RegisterCodeFix(codeActionRemoveIdentityColumn, context.Diagnostics);
+			}, context.CancellationToken);
+			
+		}
+
+		private async Task<Document> RemoveKeysFromFieldsAsync(Document document,CodeFixContext context, CancellationToken cToken, Diagnostic diagnostic, bool editIdentityAttribute = false, bool removeIdentityAttribute = false)
+		{
+
+			SemanticModel semanticModel = await document.GetSemanticModelAsync(cToken).ConfigureAwait(false);
+			PXContext pxContext = new PXContext(semanticModel.Compilation);
+			AttributeInformation attributeInformation = new AttributeInformation(pxContext);
+
+			cToken.ThrowIfCancellationRequested();
+
+			Location[] attributeLocations = diagnostic.AdditionalLocations.ToArray();
+			var identityAttributeType = pxContext.FieldAttributes.PXDBIdentityAttribute;
+			var longIdentityAttributeType = pxContext.FieldAttributes.PXDBLongIdentityAttribute;
+
+			if (attributeLocations == null)
```

Comment:
`attributeLocations` cannot be null because `ToArray()` doesn't return `null` in any case